lib/Razor2/Signature/Whiplash.pm

```diff
@@ -682,13 +685,8 @@ sub whiplash {
         # the value of length to the nearest multiple of ``length_error''.
         # Take the first 20 hex chars from SHA1 and call it the signature.
 
-        my $sha1 = Digest::SHA1->new();
-
-        $sha1->add($host);
-        $sig = substr $sha1->hexdigest, 0, 12;
-
-        $sha1->add($corrected_length);
-        $sig .= substr $sha1->hexdigest, 0, 4;
+        $sig = substr sha1_hex($host), 0, 12;
+        $sig .= substr sha1_hex($corrected_length), 0, 4;
```

Comment:
I may be missing something about the inner workings of SHA1 but to me it looks like this code behaves differently to the old one:

The old code created a SHA1 object, added the `$host`, took the first 12 characters of the digest as generated up to that point, added `$corrected_length` and then took the last four characters of the digest calculated at this new point in time.

The new code calculates two different digests, one over `$host` and one over `$corrected_length`, the latter does not include state from the previous digest.